openmc/filter.py

```diff
@@ -104,27 +104,6 @@ def __lt__(self, other):
     def __hash__(self):
         return hash(repr(self))
 
-    def __deepcopy__(self, memo):
-        existing = memo.get(id(self))
-
-        # If this is the first time we have tried to copy this object, create a copy
-        if existing is None:
-            clone = type(self).__new__(type(self))
-            clone._type = self.type
-            clone._bins = copy.deepcopy(self.bins, memo)
```

Comment:
Awesome :-) I'm probably remembering the early days when I implemented deep
copies without memo and they were very slow. Good to know that isn't a
problem for us here.
On Jun 1, 2016 12:57 PM, "Paul Romano" <notifications@github.com> wrote:

> In openmc/filter.py
> <https://github.com/mit-crpg/openmc/pull/667#discussion_r65400514>:
>
> > @@ -104,27 +104,6 @@ def __lt__(self, other):
> >      def __hash__(self):
> >          return hash(repr(self))
> >
> > -    def __deepcopy__(self, memo):
> > -        existing = memo.get(id(self))
> > -
> > -        # If this is the first time we have tried to copy this object, create a copy
> > -        if existing is None:
> > -            clone = type(self).__new__(type(self))
> > -            clone._type = self.type
> > -            clone._bins = copy.deepcopy(self.bins, memo)
>
> Yes, deepcopy operates recursively.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/mit-crpg/openmc/pull/667/files/07ef67fee019b3c2eb580fc651e6bbbf0bf2e0cb#r65400514>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAMyVAZ-NYl6I37kVl6kI55N4MhM5u0xks5qHbnpgaJpZM4IrrQU>
> .
>